christfellowshipchurch/apollos-api
# Adds the `dateTime` field to the Group Algolia Index

This PR re-enables the `dateTime` of a Group to be indexed inside of Algolia.

This field is _not_ a searchable field. Its only purpose is to store the value inside of the index so that it's more easily queried by the front-end.

To test:
Make sure you're using the Stage Algolia Index:
```
query searchGroups($query: SearchQueryInput!, $first: Int, $after: String) {
  searchGroups(query: $query, first: $first, after: $after) {
    totalResults
    pageInfo {
      startCursor
      endCursor
    }
    edges {
      id
      cursor
      title
      summary
      coverImage {
        sources {
          uri
        }
      }
      relatedNode {
        id

        ... on Url {
          url
        }

        ... on Group {
          dateTime {
            start
          }
        }
      }
      action
      ...groupSearchResultNodeFragment
    }
  }
}

fragment groupSearchResultNodeFragment on GroupSearchResult {
  preferences
  subPreferences
  meetingDay
  meetingType
  campusName
  leaders {
    firstName
    lastName
    photo {
      uri
    }
  }
}
```

## Closes Ticket
[CFDP-1637]

[CFDP-1637]: https://christfellowshipchurch.atlassian.net/browse/CFDP-1637
